extensions/binding/org.eclipse.smarthome.binding.weatherunderground/src/main/java/org/eclipse/smarthome/binding/weatherunderground/handler/WeatherUndergroundHandler.java

```diff
@@ -237,6 +215,173 @@ private void updateChannel(String channelId) {
+
+        String channelTypeId = getChannelTypeId(channelId);
+        switch (channelTypeId) {
+            case "forecastTime":
+                return undefOrState(dayForecast.getForecastTime(), new DateTimeType(dayForecast.getForecastTime()));
+            case "conditions":
+                return undefOrState(dayForecast.getConditions(), new StringType(dayForecast.getConditions()));
+            case "minTemperature":
+                return undefOrQuantity(dayForecast.getMinTemperatureC(), Units.CELSIUS, getTemperatureConversionMap());
+            case "maxTemperature":
+                return undefOrQuantity(dayForecast.getMaxTemperatureC(), Units.CELSIUS, getTemperatureConversionMap());
+            case "relativeHumidity":
+                return undefOrDecimal(dayForecast.getRelativeHumidity());
+            case "probaPrecipitation":
+                return undefOrDecimal(dayForecast.getProbaPrecipitation());
+            case "precipitationDay":
+                return undefOrQuantity(dayForecast.getPrecipitationDayMm(), MILLI(Units.METRE),
+                        getConversionMap(MILLI(Units.METRE), SmartHomeUnits.INCH));
+            case "snow":
+                return undefOrQuantity(dayForecast.getSnowCm(), CENTI(Units.METRE),
+                        getConversionMap(CENTI(Units.METRE), SmartHomeUnits.INCH));
+            case "maxWindDirection":
+                return undefOrState(dayForecast.getMaxWindDirection(),
+                        new StringType(dayForecast.getMaxWindDirection()));
+            case "maxWindDirectionDegrees":
+                return undefOrDecimal(dayForecast.getMaxWindDirectionDegrees());
+            case "maxWindSpeed":
+                return undefOrQuantity(dayForecast.getMaxWindSpeedKmh(), Units.KILOMETRE_PER_HOUR,
+                        getSpeedConversionMap());
+            case "averageWindDirection":
+                return undefOrState(dayForecast.getAverageWindDirection(),
+                        new StringType(dayForecast.getAverageWindDirection()));
+            case "averageWindDirectionDegrees":
+                return undefOrDecimal(dayForecast.getAverageWindDirectionDegrees());
+            case "averageWindSpeed":
+                return undefOrQuantity(dayForecast.getAverageWindSpeedKmh(), Units.KILOMETRE_PER_HOUR,
+                        getSpeedConversionMap());
+            case "icon":
+                State icon = HttpUtil.downloadImage(dayForecast.getIcon().toExternalForm());
+                if (icon == null) {
+                    logger.debug("Failed to download the content of URL {}", dayForecast.getIcon().toExternalForm());
+                    icon = UnDefType.UNDEF;
+                }
+                return icon;
+            default:
+                return null;
+        }
+    }
+
+    private State undefOrState(Object value, State state) {
+        return value == null ? UnDefType.UNDEF : state;
```

Comment:
Why do you return UNDEF instead of null? As far as I can see, the original code only sent state updates if there was some value available.

Reply:
The original code does also convert `null` value to `UNDEF`, see https://github.com/eclipse/smarthome/blob/ref-0.9.0/extensions/binding/org.eclipse.smarthome.binding.weatherunderground/src/main/java/org/eclipse/smarthome/binding/weatherunderground/handler/WeatherUndergroundHandler.java#L209

I guess we do want this behaviour when a value changes from _present_ to _absent_.

Reply:
Well, that's a bug in the old code then. In any case, if the binding wants to say that no state exists anymore at all (something like: The earth has just been demolished, so providing a temperature for it does not make sense anymore), it needs to set it to NULL. If it just cannot provide a current value, it must not send any state update.